src/libertem/io/dataset/hdf5.py

```diff
@@ -160,16 +160,26 @@ def _do_detect():
         # try to guess the hdf5 dataset path:
         try:
             datasets = executor.run_function(_get_datasets, path)
-            largest_ds = sorted(datasets, key=lambda i: i[1], reverse=True)[0]
-            name, size, shape, dtype = largest_ds
+            datasets_list = sorted(datasets, key=lambda i: i[1], reverse=True)
+            dataset_paths = ", ".join([ds_path[0] for ds_path in datasets_list])
```

Comment:
Why is `dataset_paths` not a list? Using a single string is problematic, as a "dataset name" in HDF5 can contain special characters:

```python
In [2]: import h5py                                                                                                                                             
In [3]: f = h5py.File("test.h5", "w")                                                                                                                           
In [4]: f.create_dataset("a,b,c", shape=(16, 16), dtype="float32")                                                                                              
Out[4]: <HDF5 dataset "a,b,c": shape (16, 16), type "<f4">
```

Now your dropdown contains three entries...

All other `detect_params` changes look good :+1:

Reply:
Fixed!